### PR TITLE
Detect bidir 5665 v14

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5665

Describe changes:
- allows bidirectional signature matching !

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1922

https://github.com/OISF/suricata/pull/11323 with better doc

First version not a draft.
Should I squash all the commits together ?

### CODE REVIEW

File: doc/userguide/rules/intro.rst [hunk 1]
```diff
@@ -248,10 +251,48 @@ Now, let's say we have a rule with the following header::
...
+
+    alert http any any => 5.6.7.8 80 (msg:"matching json to server and xml to client"; sid: 1;  :example-rule-emphasis:`bidir.toserver;` http.content_type; content: "json";  :example-rule-emphasis:`bidir.toclient;` http.content_type; content: "xml";)
+
+Bidirectional rules have some limitations :
+- They are only meant to work on transactions with first a request to the server,
+and then a response to the client, and not the other way around (not tested).
+- They cannot have ``fast_pattern`` or ``prefilter`` on a keyword which is on
+the direction to client only if they do have any streaming buffers for detection
+on the other side.
+- They will refuse to load if a single directional rule is enough.
```
Comment (jufajardini) [inline doc/userguide/rules/intro.rst:294]: When compiling the documents locally, I noticed that this won't render as a bulleted list. This should fix it:
```suggestion
Bidirectional rules have some limitations :

- They are only meant to work on transactions with first a request to the server,
  and then a response to the client, and not the other way around (not tested).
- They cannot have ``fast_pattern`` or ``prefilter`` on a keyword which is on
  the direction to client only if they do have any streaming buffers for detection
  on the other side.
- They will refuse to load if a single directional rule is enough.
```

File: doc/userguide/rules/intro.rst [hunk 2]
```diff
@@ -248,10 +251,48 @@ Now, let's say we have a rule with the following header::
...
+.. container:: example-rule
+
+    alert http any any => 5.6.7.8 80 (msg:"matching json to server and xml to client"; sid: 1;  :example-rule-emphasis:`bidir.toserver;` http.content_type; content: "json";  :example-rule-emphasis:`bidir.toclient;` http.content_type; content: "xml";)
+
+Bidirectional rules have some limitations :
+- They are only meant to work on transactions with first a request to the server,
+and then a response to the client, and not the other way around (not tested).
+- They cannot have ``fast_pattern`` or ``prefilter`` on a keyword which is on
+the direction to client only if they do have any streaming buffers for detection
+on the other side.
```
Comment (jufajardini) [inline doc/userguide/rules/intro.rst:293]: This part is still difficult to know how to read, to me. What about, maybe...
```suggestion
- They cannot have ``fast_pattern`` or ``prefilter`` on a keyword which is on
the direction to client - only if they do have any streaming buffers for detection
on the other side.
```
(If that's what this means, ofc XD )

**Review by jufajardini — CHANGES_REQUESTED** [review]
Two documentation suggestions :P

Thanks for including the explanation about how Suricata interprets `<>` rules!

Following up on the previous PR comment about name suggestions, what about transaction rule? (or something that brought attention to this aspect)

**Comment by jufajardini** [comment]
To me, it looks like all the commits could be squashed, indeed.  Or maybe have one for code, and one for docs...  🤔

File: doc/userguide/rules/intro.rst (hunk 1 above)
Comment (catenacyber) [reply doc/userguide/rules/intro.rst:294]: thanks

File: doc/userguide/rules/intro.rst (hunk 2 above)
Comment (catenacyber) [reply doc/userguide/rules/intro.rst:293]: adding an example of an impossible rule

**Comment by catenacyber** [comment]
Continued in https://github.com/OISF/suricata/pull/11528

File: doc/userguide/rules/intro.rst (hunk 1 above)
Comment (catenacyber) [reply doc/userguide/rules/intro.rst:294]: Actually your proposal does not work :-/
> /__w/suricata/suricata/doc/userguide/rules/intro.rst:290:Unexpected indentation.

Comment (jufajardini) [reply doc/userguide/rules/intro.rst:294]: hm, sorry about that, I tested locally and it worked. I possibly missed something when bringing from local changes to the github suggestion interface :/

**Pull request closed without merging**